diesel/src/query_builder/upsert/on_conflict_clause.rs

```diff
@@ -3,41 +3,62 @@ use super::on_conflict_target::*;
 use crate::backend::sql_dialect;
 use crate::backend::Backend;
 use crate::insertable::*;
+use crate::query_builder::where_clause::NoWhereClause;
 use crate::query_builder::*;
 use crate::result::QueryResult;
 
 #[doc(hidden)]
 #[derive(Debug, Clone, Copy)]
-pub struct OnConflictValues<Values, Target, Action> {
+pub struct OnConflictValues<Values, Target, Action, WhereClause> {
```

Comment:
```suggestion
pub struct OnConflictValues<Values, Target, Action, WhereClause = NoWhereClause> {
```

That makes it backward compatible with the old definition